File: assignements/C++/switches_puzzle.cpp
#include <iostream>
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

// Take input of length of array and input of combinations and figure how many switches would be needed

using namespace std;

int main(){
	int length, comb, exp_comb;
	float switch_needed;
	cout << "Input the length of array"  << endl ;
	cin >> length;
        comb = pow(2,length) ;
	printf ("max combination that we can have with array of lenth  %d is %d \n",length, comb);
	cout << "Enter the number of combination you want to find the number of switches needed" << endl;
	cin >> exp_comb;
	switch_needed = (log(exp_comb)/log(2));
	cout << ceil(switch_needed) << endl;
	return 0;
}

File: assignements/python/switches_puzzle.py
#!/usr/bin/python
from math import ceil,log

if __name__ == '__main__':
        print "Input the number of switches to find combination"
        num_switches = int(raw_input())
        comb = pow(2,num_switches)
        print "combination is : " + str(comb)
        print "Enter the combination to find the number of the switches "
        new_comb = int(raw_input())
        print "Number of switches needed is : "  + str(ceil(log(new_comb,2)))

